From 66906736fe91edf0e8b72006ac05f46546dabe6a Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Tue, 21 May 2013 17:45:28 -0400
Subject: [PATCH] nohz: print out actual name of softirq pending

We already have an array of string names for the various
softirq bits, so here we make use of them to get the name
instead of forcing people to do a manual translation.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/kernel/time/tick-sched.c b/kernel/time/tick-sched.c
index d7b8718..360619e 100644
--- a/kernel/time/tick-sched.c
+++ b/kernel/time/tick-sched.c
@@ -734,8 +734,14 @@ static bool can_stop_idle_tick(int cpu, struct tick_sched *ts)
 		static int ratelimit;
 
 		if (ratelimit < 10) {
-			pr_warn("NOHZ: local_softirq_pending %02x\n",
-				(unsigned int) local_softirq_pending());
+			u32 vec_nr;
+			u32 bits = local_softirq_pending();
+
+			pr_warn("NOHZ: local_softirq_pending (%02x)", bits);
+			for (vec_nr = 0; bits; vec_nr++, bits >>= 1)
+				if (bits & 1)
+					printk(" %s", softirq_to_name[vec_nr]);
+			printk("\n");
 			ratelimit++;
 		}
 		return false;
-- 
1.8.3.1

